### ISSUE TITLE
Basher package

### ISSUE THREAD

**Issue by Utumno** [wrye-bash/wrye-bash#163]
Basher must be split to a package. Subgoals:
- [x] de-wx'ing -> see #190
- Better and cleaner mechanism for defining and loading tabs
- [x] ~~balt~~ gui package -> see #190
- [x]  Link API #174 
- [x]  Details panels #293 

306:
- [x] _No other module should import from basher_. Only bash.py imports now - ~~still needs some review: https://github.com/wrye-bash/wrye-bash/commit/3fcbe3f87a99bca3f6aa85481697c0e3b073d390~~ - well, reviewed here: https://github.com/wrye-bash/wrye-bash/commit/e7f9493637d1c0a64877093e8017b6f1cdbd1849
- [x] column sort indicators in ModList must be moved to UIList
- [x] wx.Dialogs common superclass
- [x] sashPos: move to SashPanel - this could help bin a couple `__init__()` plus deprecate TankData params
- [x] colReverse: bin Tank variation and _setColumnReverse
- [x] bosh.links belongs to basher
- Boilerplate/duplicate code:
  - [x] `AppendToMenu` - not only huge amount of boilerplate that mixes UI in but also a huge source of wx usages.
- [x] balt.Tank and basher.List common superclass (~~Tank.gList and List.list (list !?) must be unified~~ done in favor of gList, List.list dropped)
  - [x] ~~Consider registering some not widely used callbacks like OnChar, OnDClick via hasattr calls~~ - hasattr is evil but not totally (or is it ?) --> it is.
  - [x] Installers glitches - comments~~, scroll position~~
  - [x] icons
- [x] UIList: turn style init parameters to static fields - thus binning kwargs - cf editLabels use in callbacks

307:
- Boilerplate/duplicate code:
  - [x]  Mod_Import - common Execute as in Mod_Export (also use Pycharm's search for duplicates) - done: a28fcfce1f59482d1e701504b738a8df26a9bfe4
-  UIList:
  - [x]  Erase Tank - this has two parts (edit: done in daafdd5e76746833afe4eba496aa4afac41ff439)
    - setUI: hash up an API for setUI and mouse texts based on the ex TankData API - some good stuff there but needs _thought_ - move this to UIList (edit: da640842f06132e7388953a177107037d23b891c)
    - details API - move this to UIList - actually this depends on splitting the Installers details to a class and giving details to screens and inis -> depends on  #293
  - [x] erase TankPanels etc -> depends on  #293
  - [x] make `_gList` private (`__gList`)
- **eliminate `Link.Frame` uses in bosh**: 
  - [x] modList (yak!): 6452bcb4b44d4b434e8636b35b69351cb6742869 (edit: done in 255357090bc99993fd07077c87a0656cb3a3b17f)
  - [X] rest of the uses are dialogs (askYes etc) - needs thought - add the dialog instance as method parameter ? (done in 6659eb3a977dfc8ba7210f0879020f4b07c11dc0 - yep, callbacks)
- [x] hasattr calls in NotebookPanel OnShow and ClosePanel must go

307+:
- [x] ~~UIList make self.data_store private (around 40 uses of self.window.data_store in Links)~~ would be nice to have a property though
- [x] _shellUI should be removed and replaced by event.CmdDown -> 301e46093266d8d7a34001a2f04b3c0a858f5576
- [ ] tooltips in status bar
- Dialog enhancements:
  - [ ] Esc should work on all Bash dialogs that have a cancel button
  - [ ] Go through usages of balt.button and set default=True to all dialogs for the default button (see 9a77bfc42c83c20ca9c7c06f23ba90588dfe9b1f)
- [x] Modchecker leaks ? 28d290b11f19c99f988d8e6038bc628968296281 - done in 531679d37d6c50cfc030b9c462f44250bcfab7a0

**Comment by Utumno** [comment]
Put to the backburner for now.
